### PR TITLE
dotCMS/core#15296 JavaScript Event calls need to fail more gracefully

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
-    constructor(private protocol: string, private baseUrl: string, private endPoint: string) {}
+    constructor(private url: string, private useSSL: boolean) {
 
-    get url(): string {
-        return `${this.protocol}://${this.baseUrl}${this.endPoint}`;
     }
 
-    public getHttpUrl(): string {
-        return new DotEventsSocketURL(this.protocol === 'ws' ? 'http' : 'https', this.baseUrl, this.endPoint).url;
+    public getWebSocketURL(): string {
```
Comment (fmontes) [inline projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts]: doc?

File: projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
 export class DotEventsSocketURL {
```
Comment (fmontes) [inline projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts]: doc?

File: projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts]: add `@returns ` param

File: projects/dotcms-js/src/lib/core/dotcms-events.service.ts [hunk 3]
```diff
@@ -88,4 +83,11 @@ export class DotcmsEventsService {
 
         return subject.asObservable();
     }
+
+    /**
+     * Listen  when the socket is opened
+     */
+    open(): Observable<boolean> {
```
Comment (alfredo-dotcms) [inline projects/dotcms-js/src/lib/core/dotcms-events.service.ts]: add `@returns ` param to doc

File: projects/dotcms-js/src/lib/core/util/dot-event-socket.ts [hunk 4]
```diff
@@ -32,24 +42,24 @@ export class DotEventsSocket {
...
     /**
      * Connect to a Event socket using  Web Socket protocol,
      * if a Web Socket connection can be stablish then try again with a Long Polling connection.
      */
-    connect(): void {
-        this.loggerService.debug('Creating a new socket connection', this.dotEventsSocketURL.url);
-
-        this.status = ConnectionStatus.CONNECTING;
-        this.connectProtocol();
+    connect(): Observable<ConfigParams> {
```
Comment (alfredo-dotcms) [inline projects/dotcms-js/src/lib/core/util/dot-event-socket.ts]: add `@returns ` param to doc

File: projects/dotcms-js/src/lib/core/util/dot-event-socket.ts [hunk 5]
```diff
@@ -78,6 +88,24 @@ export class DotEventsSocket {
         return this._open.asObservable();
     }
 
+    /**
+     * Return true if the socket is connected otherwise return false
+     */
+    isConnected(): boolean {
```
Comment (alfredo-dotcms) [inline projects/dotcms-js/src/lib/core/util/dot-event-socket.ts]: add `@returns ` param to doc

File: projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts [hunk 6]
```diff
@@ -1,11 +1,30 @@
...
+     * Return the web socket url to connect with the Event end point
+     */
+    public getWebSocketURL(): string {
+        return `${this.getWebSocketProtocol()}://${this.url}`;
+    }
+
+    /**
+     * Return the long polling url to connect with the Event end point
+     */
+    public getLongPoolingURL(): string {
```
Comment (alfredo-dotcms) [inline projects/dotcms-js/src/lib/core/util/models/dot-event-socket-url.ts]: add `@returns ` param to doc

**Pull request merged**